Source/player.cpp

```diff
@@ -367,27 +367,74 @@ bool PlayerStruct::IsWalking() const
 	}
 }
 
-void SetPlayerGPtrs(byte *pData, byte **pAnim)
+void PlayerStruct::Reset()
 {
-	int i;
+	// Create empty default initialized PlayerStruct on heap to avoid excessive stack usage
+	auto emptyPlayer = std::make_unique<PlayerStruct>();
+	*this = std::move(*emptyPlayer);
+}
 
-	for (i = 0; i < 8; i++) {
-		pAnim[i] = CelGetFrameStart(pData, i);
+int PlayerStruct::GetAnimationWidth(player_graphic graphic)
+{
+	switch (graphic) {
+	case player_graphic::Stand:
+		if (_pClass == HeroClass::Monk)
+			return 112;
+		return 96;
+	case player_graphic::Walk:
+		if (_pClass == HeroClass::Monk)
+			return 112;
+		return 96;
+	case player_graphic::Attack:
+		if (_pClass == HeroClass::Monk)
+			return 130;
+		if (_pClass == HeroClass::Warrior || _pClass == HeroClass::Barbarian) {
+			auto gn = static_cast<anim_weapon_id>(_pgfxnum & 0xF);
+			if (gn == ANIM_ID_BOW)
+				return 96;
+		}
+		return 128;
+	case player_graphic::Hit:
+		if (_pClass == HeroClass::Monk)
+			return 98;
+		return 96;
+	case player_graphic::Lightning:
+	case player_graphic::Fire:
+	case player_graphic::Magic:
+		if (_pClass == HeroClass::Monk)
+			return 114;
+		if (_pClass == HeroClass::Sorcerer)
+			return 128;
+		return 96;
+	case player_graphic::Death:
+		if (_pClass == HeroClass::Monk)
+			return 160;
+		return 128;
+	case player_graphic::Block:
+		if (_pClass == HeroClass::Monk)
+			return 98;
+		return 96;
+	default:
+		Log("GetAnimationWidth: Unkown graphic {}", graphic);
+		return 96;
 	}
 }
 
-void LoadPlrGFX(int pnum, player_graphic gfxflag)
+void SetPlayerGPtrs(const char *path, std::unique_ptr<byte[]> &data, std::array<std::optional<CelSprite>, 8> &anim, int width)
+{
+	data = std::move(LoadFileInMem(path));

```

Comment:
Set `data = nullptr;` first to reduce the maximum memory usage.

Reply:
No need for `std::move` here, it's an rvalue

Reply:
Done 🙂